core/dispute_coordinator/provisioner/impl/prioritized_selection.cpp

```diff
@@ -148,7 +148,7 @@ namespace kagome::dispute {
 
     for (size_t i = 0; i < disputes.size();) {
       auto batch_size = std::min(kVotesSelectionBatchSize, disputes.size() - i);
-      auto batch = gsl::span(&disputes[i], batch_size);
+      auto batch = std::span(disputes).first(batch_size);
```

Comment:
Bug 1: `.first -> 0..4 0..4 0..4` vs `.subspan -> 0..4 5..8 9..12`
Bug 2: `.first`/`.last`/`.subspan` don't check length, out of bound access
```suggestion
      auto batch = std::span(disputes).subspan(i, std::min(batch_size, disputes.size() - i));
```